### PR TITLE
gitAndTools.gitui: 0.10.1 -> 0.11.0

### PR DESCRIPTION
**Pull request by yanganto**
###### Motivation for this change
Update `gitui`, because one of the new features is required and implemented by myself

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/version-management/git-and-tools/default.nix [hunk 1]
```diff
@@ -201,6 +201,7 @@ let
 
   gitui = callPackage ./gitui {
     inherit (darwin.apple_sdk.frameworks) Security AppKit;
+    inherit (pkgs) openssl perl;
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/git-and-tools/default.nix:204]: Is this really required? None of the other packages do that.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107401` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gitAndTools.gitui</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107401` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gitAndTools.gitui</li>
  </ul>
</details>

File: pkgs/applications/version-management/git-and-tools/default.nix (hunk 1 above)
Comment (yanganto) [reply pkgs/applications/version-management/git-and-tools/default.nix:204]: Yes, it is in the `openssl-sys` build script, and I am not trace the who need `openssl-sys`.

**Comment by yanganto** [comment]
Hi @SuperSandro2000 

I try several times with `nixpkgs-review` and the return code is still `0`, may you provide more about this.  Thanks.
```bash
$ nixpkgs-review pr -p gitAndTools.gitui 107401
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/107401/head:refs/nixpkgs-review/1
$ git worktree add /home/yanganto/.cache/nixpkgs-review/pr-107401-4/nixpkgs b67ba0bfcc714453cdeb8d713e35751eb8b4c8f4
Preparing worktree (detached HEAD b67ba0bfcc7)
HEAD is now at b67ba0bfcc7 Merge pull request #107413 from 0x4A6F/master-dasel-upgrade
$ git merge --no-commit 044cd65ef0559481eeb1750a0335bb9445fdc130
Automatic merge went well; stopped before committing as requested
warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
$ nix --experimental-features nix-command build --no-link --keep-going --option build-use-sandbox relaxed -f /home/yanganto/.cache/nixpkgs-review/pr-107401-4/build.nix
warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
warning: ignoring the user-specified setting 'sandbox', because it is a restricted setting and you are not a trusted user

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/107401

1 package built:
gitAndTools.gitui

warning: ignoring the user-specified setting 'experimental-features', because it is a restricted setting and you are not a trusted user
$ nix-shell /home/yanganto/.cache/nixpkgs-review/pr-107401-4/shell.nix

[nix-shell:~/.cache/nixpkgs-review/pr-107401-4]$ echo $?
0
```

File: pkgs/applications/version-management/git-and-tools/default.nix (hunk 1 above)
Comment (yanganto) [reply pkgs/applications/version-management/git-and-tools/default.nix:204]: Here is the dependency tree with `openssl`

>>>
gitui v0.11.0 (...)
├── anyhow v1.0.36
├── asyncgit v0.11.0 (...)
│   ├── git2 v0.13.12
│   │   ├── bitflags v1.2.1
│   │   ├── libc v0.2.81
│   │   ├── libgit2-sys v0.12.14+1.1.0
│   │   │   ├── libc v0.2.81
│   │   │   ├── libssh2-sys v0.2.20
│   │   │   │   └── openssl-sys v0.9.58
│   │   │   │       └── libc v0.2.81
│   │   │   └── openssl-sys v0.9.58 (*)
│   │   ├── openssl-sys v0.9.58 (*)
>>>

Comment (yanganto) [reply pkgs/applications/version-management/git-and-tools/default.nix:204]: Here is the error if the `openssl` and `perl` remove
``` 
error: failed to run custom build command for `openssl-sys v0.9.58`

Caused by:
  process didn't exit successfully: `/build/source/target/release/build/openssl-sys-7a548c8246aa8cd3/build-script-main` (exit code: 101)
  --- stdout
  cargo:rustc-cfg=const_fn
  cargo:rerun-if-env-changed=X86_64_UNKNOWN_LINUX_GNU_OPENSSL_NO_VENDOR
  X86_64_UNKNOWN_LINUX_GNU_OPENSSL_NO_VENDOR unset
  cargo:rerun-if-env-changed=OPENSSL_NO_VENDOR
  OPENSSL_NO_VENDOR unset
  CC_x86_64-unknown-linux-gnu = Some("/nix/store/b96dqbx6pri2xp2xxlq6i269virrdaw6-gcc-wrapper-9.3.0/bin/cc")
  CFLAGS_x86_64-unknown-linux-gnu = None
  CFLAGS_x86_64_unknown_linux_gnu = None
  HOST_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  running "perl" "./Configure" "--prefix=/build/source/target/x86_64-unknown-linux-gnu/release/build/openssl-sys-9a6c36981fda3786/out/openssl-build/install" "no-dso" "no-ssl3" "no-unit-test" "no-comp" "no-zlib" "no-zlib-dynamic" "no-md2" "no-rc5" "no-weak-ssl-ciphers" "no-camellia" "no-idea" "no-seed" "no-shared" "linux-x86_64" "-O2" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64"

  --- stderr
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /build/gitui-0.11.0-vendor.tar.gz/openssl-src/src/lib.rs:411:39
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: build failed
builder for '/nix/store/s9ld60ajf2pka8azyycsh6myykjs86dk-gitui-0.11.0.drv' failed with exit code 101
error: build of '/nix/store/s9ld60ajf2pka8azyycsh6myykjs86dk-gitui-0.11.0.drv' failed
```

Comment (SuperSandro2000) [reply pkgs/applications/version-management/git-and-tools/default.nix:204]: I mean why can'T we put openssl and perl as a normal input and inherit it?

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107401` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gitAndTools.gitui</li>
  </ul>
</details>

Might be a flaky test...

File: pkgs/applications/version-management/git-and-tools/default.nix (hunk 1 above)
Comment (yanganto) [reply pkgs/applications/version-management/git-and-tools/default.nix:204]: Thanks for reviewing on this.  I will take care on this when `gitui` go to 0.12.  Thanks.

Comment (SuperSandro2000) [reply pkgs/applications/version-management/git-and-tools/default.nix:204]: You are sure you will remember this? I am not.

Comment (yanganto) [reply pkgs/applications/version-management/git-and-tools/default.nix:204]: I'll remember this. No worry.

**Pull request merged**